### PR TITLE
Rename "group name" into "cluster name"

### PR DESCRIPTION
**Pull request by petrpleshachkov**
Moved (and renamed) groupName and groupPassword to the Config/ClientConfig
configuration classes. Removed GroupConfig configuration class.

EE part: https://github.com/hazelcast/hazelcast-enterprise/pull/3157

### CODE REVIEW

**Comment by petrpleshachkov** [comment]
run-lab-run

**Comment by petrpleshachkov** [comment]
run-lab-run

**Comment by petrpleshachkov** [comment]
run-lab-run

**Comment by petrpleshachkov** [comment]
run-lab-run

**Comment by petrpleshachkov** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientTransactionManagerServiceImpl.java [hunk 1]
```diff
@@ -103,12 +102,12 @@ public void shutdown() {
     }
 
     @Override
-    public String getGroupName() {
-        final GroupConfig groupConfig = client.getClientConfig().getGroupConfig();
-        if (groupConfig == null) {
-            throw new RuntimeException("GroupConfig cannot be null client is participate in XA Transaction");
+    public String getClusterName() {
+        final String clusterName = client.getClientConfig().getClusterName();
+        if (clusterName == null) {
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientTransactionManagerServiceImpl.java]: how can a `clusterName` be `null`?

**Comment by ahmetmircik** [comment]
@petrpleshachkov do we need a counterpart PR for management center?

File: hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientTransactionManagerServiceImpl.java (hunk 1 above)
Comment (petrpleshachkov) [reply hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientTransactionManagerServiceImpl.java]: I will remove this `if-block` which is a leftover from old code.

**Comment by petrpleshachkov** [comment]
run-lab-run

**Comment by kwart** [comment]
I've discussed the changes with @petrpleshachkov . We'll merge this after a rebase to latest master. I'll rebase my own changes afterward.

**Pull request merged**